proxy/tests/support/mod.rs

```diff
@@ -30,6 +30,48 @@ use self::tower_h2::{Body, RecvBody};
 use std::net::SocketAddr;
 pub use std::time::Duration;
 
+/// Environment variable for overriding the test patience.
+pub const ENV_TEST_PATIENCE_MS: &'static str = "RUST_TEST_PATIENCE_MS";
+pub const DEFAULT_TEST_PATIENCE: u64 = 15;
+
+#[macro_export]
+macro_rules! eventually {
+    ($cond:expr, retries: $retries:expr, $($arg:tt)+) => {
+        {
+            use std::{env, u64};
+            use std::time::{Instant, Duration};
+            use std::str::FromStr;
+            // TODO: don't do this *every* time eventually is called (lazy_static?)
```

Comment:
Probably... but not super important.

Reply:
The main reason I didn't just go ahead and use `lazy_static` is because it's a macro, and `extern crate` statements with `#[macro_use]` have to go in the crate root. Since `tests/support` is a module that's included separately in each test module, that means that each test file would have to `#[macro_use] extern crate lazy_static;` or else it won't compile, even if it doesn't actually make use of the `assert_eventually!` macro. I thought this behaviour would be confusing and might pose a minor barrier to entry for writing new tests.

WDYT?

Reply:
Yea, re-checking the environment is fine, it's just a test, and that way we don't need the lazy_static dependency.